app/views/assets/_stream_bed_profile.html.haml

```diff
@@ -27,16 +27,11 @@
           .row
             .col-sm-12
               #streambed_table_actions.btn-toolbar
-                - if can?(:update, @asset) && can?(:manage, StreambedProfile) && !@asset.assigned_version
+                - if can?(:update_from_structure, @asset) && !@asset.assigned_version && @asset.streambed_profiles.map{|p| p.inspection&.state}.any?('final')
                   .btn-group
-                    = link_to new_streambed_profile_path(transam_asset_id: @asset.object_key), remote: true, class: 'btn btn-primary', title: 'New Profile' do
+                    = link_to new_streambed_profile_streambed_profile_point_path(@asset.streambed_profiles.map{|p| p.inspection&.state == "final" ? p : nil}.select{|p| !p.nil?}.last), remote: true, class: 'btn btn-primary', title: 'New Distance' do
```

Comment:
`@asset.streambed_profiles.map{|p| p.inspection&.state == "final" ? p : nil}.select{|p| !p.nil?}.last` could be more simply implemented as `@asset.streambed_profiles.select{|p| p.inspection&.state == "final"}.last`.